nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -185,30 +259,7 @@ def test_signals_extraction_with_labels():
     signals_r, labels_r = signal_extraction.img_to_signals_labels(
         data_img, labels_img, mask_img=mask_img)
     np.testing.assert_almost_equal(signals_r, signals)
-    assert labels_r == list(range(1, 9))
-
-    # Test input validation
-    data_img = nibabel.Nifti1Image(np.zeros((2, 3, 4, 5)), np.eye(4))
-
-    good_labels_img = nibabel.Nifti1Image(np.zeros((2, 3, 4)), np.eye(4))
-    bad_labels1_img = nibabel.Nifti1Image(np.zeros((2, 3, 5)), np.eye(4))
-    bad_labels2_img = nibabel.Nifti1Image(np.zeros((2, 3, 4)), 2 * np.eye(4))
-
-    good_mask_img = nibabel.Nifti1Image(np.zeros((2, 3, 4)), np.eye(4))
-    bad_mask1_img = nibabel.Nifti1Image(np.zeros((2, 3, 5)), np.eye(4))
-    bad_mask2_img = nibabel.Nifti1Image(np.zeros((2, 3, 4)), 2 * np.eye(4))
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
-                  data_img, bad_labels1_img)
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
-                  data_img, bad_labels2_img)
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
-                  data_img, bad_labels1_img, mask_img=good_mask_img)
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
-                  data_img, bad_labels2_img, mask_img=good_mask_img)
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
-                  data_img, good_labels_img, mask_img=bad_mask1_img)
-    pytest.raises(ValueError, signal_extraction.img_to_signals_labels,
```

Comment:
I'm wondering whether we should delete these tests or not... :thinking: 
You have added tests for the helper functions you created which handle the input checking, but these tests make sure these functions are called as expected from `img_to_signals_labels` when providing bad inputs to this function.